index.rst

```diff
@@ -5,77 +5,331 @@
 
+Another alternative is to implement a template system, where the user writes a template report with placeholders for text and images.
+Then, the report can be generated by executing code to produce the placeholder values and images and exporting the report.
+The report could be a webpage, a pdf (e.g. generated with LaTeX) or any other electronic format.
 
 Service Architecture
 ====================
 
-The Catcher would be a service built on top of RSP functionalities (some existing and others still under development), with ancillary support from LOVE and OCPS.
-It will mostly rely on `Flux`_ scripts and `InfluxDB task`_ to execute operations based on data streams from the EFD bucket.
-The results can be persisted in the EFD, along with summary information to be displayed by LOVE.
+In summary our proposal is:
+
+*  The ``Catcher`` will be an indexed CSC where each instance handle a single rule.
+*  Rules will adopt the "configuration-based" approach.
+   The ``Catcher`` codebase will have a series of "simple" rules that can be chained together to form an activation flag.
+   The rule is only activated if all conditions are met.
+*  When a rule is activated, it executes an analysis process.
+   The analysis process should generate information about alarm level and data to assemble a report.
+*  A report is generated using the output of the analysis process.
+*  An alarm is issued indicating the severity level and information about how to access the report.
+*  Whether a report is generated and/or an alarm is published every time the rule is activated or only at certain severity levels will be a configuration parameter.
+
+Interface
+---------
+
+The following is an initial proposal for the (xml) interface of the ``Catcher`` and may be revised later.
+
+Commands
+^^^^^^^^
+
+``mute``
+
+   Mute this instance of the ``Catcher``.
+   It will continue to process the data stream and generating reports, but will not issue new alarm unless they have severity higher than the one specified.
+
+   Contains the following parameters:
+
+   ``severity``
+
+      Severity to mute alarms.
+      Alarms will only be issued if severity is higher than this value.
+   
+   ``duration``
+
+      How long to mute?
+   
+   ``mutedBy``
+
+      Person who muted the alarm.
+
+   ``reason``
+
+      Reason alarm is being muted.
+
+``acknowledge``
+
+   Acknowledge alarm.
+
+   Contains the following parameters:
+
+   ``acknowledgedBy``
+
+      Person acknowledging the alarm.
+
+   ``id``
+
+      Id of the process to acknowledge.
+
+``showAnalysisProcess``
+
+   Publish updated information about the analysis processes.
+
+``showAlarms``
+
+   Publish alarm information.
+
+Events
+^^^^^^
+
+``alarm``
+
+   Alarm issued when an analysis process results in a severity level above configured threshold.
```

Comment:
So does this get issued twice? Once when the first condition is met, then a second time when the artifact is ready?

Reply:
I guess i'm a bit confused why there is a `reportUrl` here but also and `analysisProcessSummary`